tests/test_on_data/test_synchronization/test_interface_synchronization.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+from numpy.testing import assert_array_equal
+from hdmf.testing import TestCase
+
+from neuroconv import ConverterPipe
+from neuroconv.datainterfaces import SpikeGLXNIDQInterface, SpikeGLXRecordingInterface, DeepLabCutInterface
+from neuroconv.tools.testing import MockBehaviorEventInterface, MockSpikeGLXNIDQInterface
+
+from ..setup_paths import ECEPHY_DATA_PATH, BEHAVIOR_DATA_PATH
+
+
+class TestNIDQSynchronization(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.unsynchronized_dlc_timestamps = np.arange(
+            start=0.0,  # First DLC pulse sent when device turns on
+            stop=17.0,  # Was acquiring for 17 seconds
+            step=1 / 30.0,  # Sent a TTL pulse every on every frame at a rate of 30 fps
+        )
+        cls.synchronized_dlc_timestamps = np.arange(
+            start=3.23,  # DLC device starts acquiring 3.23 seconds after SpikeGLX
+            stop=20.23,
+            step=1 / 30 + 0.00001,  # Synchronization reveals slight drift compared with the SpikeGLX times
+        )
```

Comment:
For conversational reference to comments above.

Note this is not actually something we've seen with DLC before, I was just using that since it's an established behavioral interface in NeuroConv.

But the setup is taken from work we did with several custom interfaces in the Fee lab: https://github.com/catalystneuro/fee-lab-to-nwb/blob/main/src/fee_lab_to_nwb/happ_ecephys/happmotifinterface.py#L55-L67

Motif timings in a MATLAB file were given in the secondary time system of the audio recorder, and the audio stream had events identified and parsed (by them, timestamps recorded in a separate MATLAB file) so we had to estimate the true motif (and subsequent syllable) times in the primary system given both the IMEC and audio times as reference

Reply:
Swapped from DLC to CsvTimeInterval's since that is actually a use case we've seen, or much closer to it anyway